python/pyarrow/array.pxi

```diff
@@ -1028,8 +1031,10 @@ cdef class Array(_PandasConvertible):
             PrettyPrintOptions options
 
         with nogil:
-            options = PrettyPrintOptions(indent, window)
+            options = PrettyPrintOptions(top_level_indent, window)
             options.skip_new_lines = skip_new_lines
+            if indent is not 0:
+                options.indent_size = indent
```

Comment:
We might want to set this always? (instead of only when it is not 0?) The default in C++ is 2 and not 0, so we will need to override that for 0 as well.

Reply:
Actually, since this is a new keyword: we probably want to follow the C++ default (which is also what is used now), and so use a default of 2 here in Python as well for the `indent` keyword

Reply:
That sounds good! I'll set the default as 2 for the moment!